wee_slack.py

```diff
@@ -3651,6 +3651,16 @@ def command_openweb(data, current_buffer, args):
         url = "https://{}/archives/{}/p{}000000".format(channel.team.domain, channel.slack_name, now.majorstr())
     w.prnt_date_tags(channel.team.channel_buffer, SlackTS().major, "openweb,logger_backlog_end,notify_none", url)
 
+@slack_buffer_required
+def command_linkarchive(data, current_buffer, args):
+    channel = EVENTROUTER.weechat_controller.buffers[current_buffer]
+    message_id = args.split(' ', 1)[1]
+    if message_id[0] == "$":
+        message_id = message_id[1:]
+    if isinstance(channel, SlackChannelCommon) and message_id in channel.hashed_messages:
+        ts = channel.hashed_messages[message_id].ts
+        url = "https://{}/archives/{}/p{}{:0>6}".format(channel.team.domain, channel.identifier, ts.majorstr(), ts.minorstr())
```

Comment:
I'm trying to figure out if it's better to use the [chat.getPermalink](https://api.slack.com/methods/chat.getPermalink) method to create this link, rather than hard code it.

On the one side, we would be sure that the link is correct, even if it changes. And links to thread messages should have some extra parameters, which this includes.

On the other side, the link is probably not going to change, we can include the thread parameters ourselves, and it would be nice to having to do a request.

Reply:
The documentation says
> This method allows hundreds of requests per minute. Use it more or less as often as required, but please obey HTTP 429 Too Many Requests status codes and wait to retry.
So I guess we can use it as much as necessary and shouldn’t think too much about it?
But yes, that means having one call each time you want to generate a link (which should be okay, I guess?)

Reply:
My concern isn't about rate limiting, it's about slow requests. Normally they will go fast enough, but what should we do if the request takes a long time for some reason? A blocking request will freeze all of weechat, so that's not acceptable. With an async request, it might look like nothing is happening, until the input field is suddenly populated later. Unless we show some loading, but ugh.

Reply:
Blocking means that it will freeze all of weechat completely if the network hangs for some reason. Even if Slack always responds fast on that request, one might have some random network issues causing requests hanging. We can never allow that to happen, so a blocking is off the table.

Let's just go for hard coding it, like you have done. But let's add the query parameters described on that page for threads.